src/Parsers/ASTRenameQuery.h

```diff
@@ -94,7 +94,24 @@ class ASTRenameQuery : public ASTQueryWithOutput, public ASTQueryWithOnCluster
     ASTPtr clone() const override
     {
         auto res = std::make_shared<ASTRenameQuery>(*this);
-        res->cloneChildren();
+        res->children.clear();
+
+        auto clone_child = [&res](ASTPtr & node)
+        {
+            if (node)
+            {
+                node = node->clone();
```

Comment:
Maybe it's me, but readability of this isn't great (will `elem.from.database` be updated? Yes, but it's not obvious). Not sure how to make it better without duplicating code though, so feel free to ignore it.